### PR TITLE
Import

### PR DESCRIPTION
**Pull request by pictureb00k**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/imprt/ImportLaunchHandlerImpl.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @Override
+    public OperationCompletionRS importLaunch(String projectId, String userName, String format, MultipartFile file) {
+        Project project = projectRepository.findOne(projectId);
+        expect(project, notNull()).verify(PROJECT_NOT_FOUND, projectId);
+
+        ImportType type = ImportType.fromValue(format);
+        expect(type, notNull()).verify(BAD_REQUEST_ERROR, type);
+
+        ImportLaunch strategy = factory.getImportLaunch(type);
+        String launch = strategy.importLaunch(projectId, userName, file);
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/imprt/ImportLaunchHandlerImpl.java]: importLaunch.imprortLaunch. Can we rename smth?

File: src/main/java/com/epam/ta/reportportal/core/imprt/impl/ImportLaunchFactoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    .build();
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public ImportLaunch getImportLaunch(ImportType type) {
+        return applicationContext.getBean(MAPPING.get(type));
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/imprt/impl/ImportLaunchFactoryImpl.java]: don't need to lookup in context. Inject instances in constructor and fill map with them

File: src/main/java/com/epam/ta/reportportal/core/imprt/impl/ImportType.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.Arrays;
+
+public enum ImportType {
+    JUNIT;
+
+    public static ImportType fromValue(String value) {
+        return Arrays.stream(ImportType.values())
+                .filter(type -> type.name().equalsIgnoreCase(value))
+                .findFirst().orElse(null);
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/imprt/impl/ImportType.java]: return Optional here

File: src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/AsyncJunitImportLaunch.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.concurrent.*;
+import java.util.function.Predicate;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+@Service
+public class AsyncJunitImportLaunch implements ImportLaunch {
+
+    @Autowired
+    private Provider<JunitParseJob> xmlParseJobProvider;
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/AsyncJunitImportLaunch.java]: do we really need it in the app context? Why can't create it with new?

File: src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/JunitImportHandler.java [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        this.launchId = launchId;
+        this.userName = user;
+        return this;
+    }
+
+    private static Date toDate(LocalDateTime startTime) {
+        return Date.from(startTime.atZone(ZoneId.systemDefault()).toInstant());
+    }
+
+    private long toMillis(String duration) {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/JunitImportHandler.java]: put it to some utilities. Like DateUtils or smth like that

File: src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/JunitImportHandler.java [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        Double value = Double.valueOf(duration) * 1000;
+        return value.longValue();
+    }
+
+    public void setProjectId(String projectId) {
+        this.projectId = projectId;
+    }
+
+    public void setUserName(String userName) {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/JunitImportHandler.java]: do we need setters here?

File: src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/AsyncJunitImportLaunch.java (hunk 4 above)
Comment (pictureb00k) [inline src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/AsyncJunitImportLaunch.java]: yes, cause if we go further into JunitParseJob we will find autowired provider for sax handler, and sax handler also contains autowired services

File: src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/AsyncJunitImportStrategy.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            throw new ReportPortalException("Problem with parsing zip file.", e);
+        }
+    }
+
+    private ParseResults processResults(CompletableFuture[] futures) {
+        ParseResults results = new ParseResults();
+        Arrays.stream(futures).map(it -> {
+            try {
+                return (ParseResults) it.get();
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/AsyncJunitImportStrategy.java]: join()

**Pull request merged**